### PR TITLE
PLU-275: [BTN-MIGRATION-4] Enable users to add Postman SMS connections

### PR DESCRIPTION
**Pull request by ogp-weeloong**
# Problem
We need to enable users to use their own agencies' sender ID when sending SMSes.

# Solution
Instead of using a Plumber default campaign, we will make the Postman SMS app support user added connections instead. This will enable them to use their agency's campaign and sender ID.

Note that we will need the user to setup the campaign template in a very specific way. We will inform them of this via the info message and link to our guide.

In the future, we can consider adding a "user-added-or-default" connection mode 

# Tests
- [TODO] Updated unit tests
- Check that I can send SMS.

### CODE REVIEW

**Comment by ogp-weeloong** [comment]
* **#625** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/625?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#607** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/607?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#616](https://github.com/opengovsg/plumber/pull/616) <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/616?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#606** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/606?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#605** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/605?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop-v2`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ogp-weeloong and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/backend/src/apps/postman-sms/actions/send-sms/index.ts [hunk 1]
```diff
@@ -17,15 +18,20 @@ const action = {
...
+   * "Template Variables" multi-select.
    *
-   * But in the future, we want to allow users to provide their own API key and
-   * specify their own campaigns and templates. We can't do this right now
-   * because Postman does not provide an API to query templates / template
-   * variables.
+   * Instead, for this action, we will instruct users to set up a campaign whose
+   * template is just a {{body}} variable. We will provide a more advanced "Send
+   * SMS with Variables" action later on, for users who are comfortable working
+   * with template variables.
```
Comment (m0nggh) [inline packages/backend/src/apps/postman-sms/actions/send-sms/index.ts:35]: question: this can only be added when postman provides the API right?

Comment (ogp-weeloong) [reply packages/backend/src/apps/postman-sms/actions/send-sms/index.ts:35]: yep yep, we can code it out for them tbh, but not a priority.

File: packages/backend/src/apps/postman-sms/auth/is-still-verified.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import { authDataSchema } from './schema'
+
+const isStillVerified: IUserAddedConnectionAuth['isStillVerified'] = async (
+  $: IGlobalVariable,
+) => {
+  const { campaignId } = authDataSchema.parse($.auth.data)
+
+  try {
+    // Use message retrieval API as a test query
+    await $.http.get('/campaigns/:campaignId/messages?limit=1', {
```
Comment (pregnantboy) [inline packages/backend/src/apps/postman-sms/auth/is-still-verified.ts:14]: gud

**Review by pregnantboy — APPROVED** [review]
lgtm after changing:
- campaign_url -> campaign_Id
- get env from api key prefix

**Comment by ogp-weeloong** [comment]
### Merge activity

* **Jun 14, 5:49 AM EDT**: @ogp-weeloong started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/625).
* **Jun 14, 5:52 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/625) rebased this pull request as part of a merge.
* **Jun 14, 5:52 AM EDT**: @ogp-weeloong merged this pull request with [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/625).

**Pull request merged**